rdata/parser/_parser.py

```diff
@@ -1284,8 +1248,10 @@ def parse_rdata_binary(
         data = data[len(format_dict[format_type]):]
 
     if format_type is RdataFormats.XDR:
+        from rdata.io.xdr import ParserXDR
+
         parser = ParserXDR(
-            data,
+            io.BytesIO(data),

```

Comment:
Shouldn't this be done inside the `init` method of the parser, instead?

Reply:
That can be done. The idea of having this line here would be that ParserXDR could process directly file objects like `f = open(...)` instead of first reading the whole file to memory outside the parser. But as nothing like that is done, I'll move this line to `init` instead.

Reply:
We could do that, but I do not think that we would gain much with it, as the tree of S-expressions is still fully constructed in memory.